pd/src/components/ibc.rs

```diff
@@ -32,11 +50,98 @@ impl Component for IBCComponent {
         Ok(())
     }
 
-    async fn execute_tx(&mut self, _tx: &Transaction) -> Result<()> {
-        Ok(())
+    async fn execute_tx(&mut self, tx: &Transaction) -> Result<()> {
+        // handle client transactions
+        for action in tx.transaction_body.actions.iter() {
+            let ibc_action = match action {
+                Action::IBCAction(ibc_action) => Some(ibc_action),
+                _ => None,
+            };
+            if ibc_action.is_none() {
+                continue;
+            }
+
+            match &ibc_action.unwrap().action {
+                // Handle IBC CreateClient. Here we need to validate the following:
+                // - client type is one of the supported types (currently, only Tendermint light clients)
+                // - consensus state is valid (is of the same type as the client type, also currently only Tendermint consensus states are permitted)
+                //
+                // Then, we compute the client's ID (a concatenation of a monotonically increasing
+                // integer, the number of clients on Penumbra, and the client type) and commit the
+                // following to our state:
+                // - client type
+                // - consensus state
+                // - processed time and height
+                CreateClient(raw_msg_create_client) => {
+                    // NOTE: MsgCreateAnyClient::try_from will validate that client_state and
+                    // consensus_state are Tendermint client and consensus states only, as these
+                    // are the only currently supported client types.
+                    let msg_create_client =
+                        MsgCreateAnyClient::try_from(raw_msg_create_client.clone())?;
+
+                    // get the current client counter
+                    let id_counter = self.client_counter().await?;
+                    let client_id =
+                        ClientId::new(msg_create_client.client_state.client_type(), id_counter)?;
+
+                    tracing::info!("creating client {:?}", client_id);
+
+                    self.store_new_client(client_id, msg_create_client).await?;
+                }
```

Comment:
What's the best way to factor this code out, so that we don't end up with all of the IBC state machine directly in the body of `execute_tx`?